src/MicroOcpp.cpp

```diff
@@ -1397,6 +1397,18 @@ void authorize(const char *idTag, OnReceiveConfListener onConf, OnAbortListener
         MO_DBG_ERR("idTag format violation. Expect c-style string with at most %u characters", IDTAG_LEN_MAX);
         return;
     }
+
+    //check for active transactions
+    for (unsigned int cId = 1; cId < context->getModel().getNumConnectors(); cId++) {
+        if (isTransactionActive(cId)) {
```

Comment:
to be honest, this is not really needed, as it's checked again in `endTransaction()`...